In [20]:
import pandas as pd

In [21]:
df = pd.read_csv('Dialysis_Facility_Compare_-_Listing_by_Facility.csv')

In [22]:
df.head()

Provider Number  Network                                  Facility Name  \
0            42592       13           DCI - LITTLE ROCK RENAL SERVICES LLC   
1            52761       18                 DAVITA-PREMIER DIALYSIS CENTER   
2            72549        1                  WALLINGFORD DIALYSIS CARE LLC   
3            82524        4                       DSI LAUREL DIALYSIS, LLC   
4            82527        4  FRESENIUS MEDICAL CARE SOUTHERN DELAWARE, LLC   

            Five Star Date Five Star  Five Star Data Availability Code  \
0  01/01/2015 - 12/31/2018         2                                 1   
1  01/01/2015 - 12/31/2018         3                                 1   
2  01/01/2015 - 12/31/2018         5                                 1   
3  01/01/2015 - 12/31/2018         4                                 1   
4  01/01/2015 - 12/31/2018         5                                 1   

                  Address Line 1                 Address Line 2         City  \
0            1910 JOHN BARROW RD                            NaN  LITTLE ROCK   
1           7612 ATLANTIC AVENUE                            NaN       CUDAHY   
2          720 N MAIN STREET EXT                        SUITE 3  WALLINGFORD   
3  LAUREL SQUARE SHOPPING CENTER  30214 SUSSEX HIGHWAY, UNIT #4       LAUREL   
4             9115 ANTIQUE ALLEY                            NaN  BRIDGEVILLE   

  State  ...  95% C.I. (lower limit) for SWR  \
0    AR  ...                             .12   
1    CA  ...                              .2   
2    CT  ...                   Not Available   
3    DE  ...                   Not Available   
4    DE  ...                   Not Available   

  Number of patients in this facility for SWR  \
0                                          56   
1                                         105   
2                                          12   
3                                           8   
4                                           1   

  Standardized First Kidney Transplant Waitlist Ratio PPPW category text  \
0                                                .45         As Expected   
1                                                .53         As Expected   
2                                      Not Available         As Expected   
3                                      Not Available         As Expected   
4                                      Not Available         As Expected   

  Patient prevalent transplant waitlist data availability code  \
0                                                  1             
1                                                  1             
2                                                  1             
3                                                  1             
4                                                  1             

  95% C.I. (upper limit) for PPPW  95% C.I. (lower limit) for PPPW  \
0                            22.8                              3.4   
1                            32.1                             11.8   
2                            74.5                             10.1   
3                            74.2                               .2   
4                            48.4                              2.8   

   Number of patients for PPPW  Percentage of Prevalent Patients Waitlisted  \
0                          100                                          9.2   
1                          209                                         20.1   
2                           24                                         36.4   
3                           19                                          7.4   
4                           44                                         14.2   

                        Location  
0    POINT (-92.372456 34.73624)  
1  POINT (-118.187206 33.966829)  
2   POINT (-72.809521 41.472838)  
3                            NaN  
4   POINT (-75.592475 38.735365)  

[5 rows x 119 columns]

In [23]:
for col in df.columns: 
    print(col) 

Provider Number
Network
Facility Name
Five Star Date
Five Star
Five Star Data Availability Code
Address Line 1
Address Line 2
City
State
Zip
County
Phone Number
Profit or Non-Profit
Chain Owned
Chain Organization
Late Shift
# of Dialysis Stations
Offers in-center hemodialysis
Offers peritoneal dialysis
Offers home hemodialysis training
Certification or Recertification Date
Claims Date
CROWNWeb Date
STrR Date
Percentage of Medicare patients with Hgb<10 g/dL
HGB<10 data availability code
Percentage of Medicare patients with Hgb>12 g/dL
Hgb > 12 data availability code
Number of Dialysis Patients with Hgb data
Patient Transfusion data availability Code
Patient Transfusion category text
Number of patients included in the transfusion summary
Percent of Adult HD patients with Kt/V >= 1.2
Adult HD Kt/V data availability code
Percentage of Adult PD PTS with Kt/V >= 1.7
Adult PD Kt/V Data Availability Code
Percentage of Pediatric HD patients with Kt/V >= 1.2
Pediatric HD Kt/V Data Availability C

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7626 entries, 0 to 7625
Columns: 119 entries, Provider Number to Location
dtypes: int64(27), object(92)
memory usage: 6.9+ MB


In [25]:
df.describe()

Provider Number      Network  Five Star Data Availability Code  \
count      7626.000000  7626.000000                       7626.000000   
mean     315876.077105     9.660372                         31.525833   
std      193257.685562     4.708672                         83.403249   
min       12306.000000     1.000000                          1.000000   
25%      142822.250000     6.000000                          1.000000   
50%      322545.500000     9.000000                          1.000000   
75%      452511.750000    14.000000                          1.000000   
max      852568.000000    18.000000                        261.000000   

                Zip   Late Shift  # of Dialysis Stations  \
count   7626.000000  7626.000000             7626.000000   
mean   50297.359691     0.167060               17.463284   
std    27581.675445     0.373054                8.495540   
min      603.000000     0.000000                0.000000   
25%    29533.000000     0.000000               12.000000   
50%    46032.000000     0.000000               17.000000   
75%    76060.750000     0.000000               22.000000   
max    99801.000000     1.000000               80.000000   

       Offers in-center hemodialysis  Offers peritoneal dialysis  \
count                    7626.000000                 7626.000000   
mean                        0.940336                    0.525439   
std                         0.236879                    0.499385   
min                         0.000000                    0.000000   
25%                         1.000000                    0.000000   
50%                         1.000000                    1.000000   
75%                         1.000000                    1.000000   
max                         1.000000                    1.000000   

       Offers home hemodialysis training  HGB<10 data availability code  ...  \
count                         7626.00000                    7626.000000  ...   
mean                             0.29006                      31.839890  ...   
std                              0.45382                      75.349864  ...   
min                              0.00000                       1.000000  ...   
25%                              0.00000                       1.000000  ...   
50%                              0.00000                       1.000000  ...   
75%                              1.00000                       1.000000  ...   
max                              1.00000                     258.000000  ...   

       Patient Hospitalization data availability Code  \
count                                      7626.00000   
mean                                         25.68673   
std                                          74.10966   
min                                           1.00000   
25%                                           1.00000   
50%                                           1.00000   
75%                                           1.00000   
max                                         258.00000   

       Patient Hospital Readmission data availability Code  \
count                                        7626.000000     
mean                                           26.699449     
std                                            74.577879     
min                                             1.000000     
25%                                             1.000000     
50%                                             1.000000     
75%                                             1.000000     
max                                           258.000000     

       Patient Survival data availability code  \
count                              7626.000000   
mean                                 30.239182   
std                                  79.229534   
min                                   1.000000   
25%                                   1.000000   
50%                                   1.000000   
75%                                   1.000000   
m

In [28]:
df['Five Star'].unique()

array(['2', '3', '5', '4', 'Not Available', '1'], dtype=object)

# Column Names Defined

### Star Rating Explained
The “Dialysis Facility Compare (DFC) Star Program” is a rating system developed by Medicare that assigns 1 to 5 stars to dialysis facilities by comparing the health of the patients in their clinics to the patients in other dialysis facilities across the country.

### Serum phosphorus significance
Serum phosphorus is measured in milligrams of phosphorus per deciliter of blood (mg/dL). According to Mayo Medical Laboratories, a normal range for adults is generally 2.5 to 4.5 mg/dL. The normal range varies slightly depending on your age.
Most commonly, a high level of phosphorus is related to a kidney disorder. It shows that your kidneys are having difficulty clearing phosphorus from your blood. A high level of phosphorus can also mean uncontrolled diabetes and other endocrine disorders.

### Concept of Kt/V
The parameter Kt/V is a measurement of the efficacy of a hemodialysis session. It identifies the effective removal of a specific solute (clearance K) resulting from a given treatment (characterized by time t) in a given patient (with a specific volume of distribution V for the solute considered).

### Hgb g/dL
A normal hemoglobin level is 11 to 18 grams per deciliter (g/dL), depending on your age and gender. But 7 to 8 g/dL is a safe level. Your doctor should use just enough blood to get to this level.

### Hypercalcemia & Serum Calcium
Hypercalcemia is a condition in which the calcium level in your blood is above normal. Too much calcium in your blood can weaken your bones, create kidney stones, and interfere with how your heart and brain work. Hypercalcemia is usually a result of overactive parathyroid glands.
A normal serum calcium level is 8-10 mg/dL (2-2.5 mmol/L) with some interlaboratory variation in the reference range, and hypercalcemia is defined as a serum calcium level greater than 10.5 mg/dL (>2.5 mmol/L).

### AV fistula
The best type of long-term access is an AV fistula. A surgeon connects an artery to a vein, usually in your arm, to create an AV fistula. An artery is a blood vessel that carries blood away from your heart. A vein is a blood vessel that carries blood back toward your heart. When the surgeon connects an artery to a vein, the vein grows wider and thicker, making it easier to place the needles for dialysis. The AV fistula also has a large diameter that allows your blood to flow out and back into your body quickly. The goal is to allow high blood flow so that the largest amount of blood can pass through the dialyzer.

### nPCR
The normalized protein catabolic rate (nPCR) is a formula commonly used to assess dietary protein intake in dialysis patients, as a means towards determining nutritional adequacy, a major problem in many ESRD patients. 
The nPCR is reported in grams of urea nitrogen per kilogram per day.
Most guidelines specify maintaining the protein intake above 1.0 – 1.2 g/kg/day in dialysis patients, with values less than 0.8 g/kg/day being equated with malnutrition.

### Standardized Waitlist Ratio (SWR)
The Standardized Waitlist Ratio (SWR) measure tracks the number of incident patients at the dialysis facility under the age of 75 listed on the kidney or kidney-pancreas transplant waitlist or who received a living donor transplant within the first year of initiating dialysis. SWR is calculated to compare the observed waitlisting rate in the facility to the waitlisting rate that was expected. 

# Preliminary thoughts for data analysis
I'm interested in comparing dialysis star rating to patient health indicators (nPCR, serum calcium, Hgb, Kt/V, serum phosphorus.
Also might be interesting to compare patient health indicators to a center's profit/ non-profit status or whether or not it's a chain buisness. 
Another idea is the relationship between star rating or profit/ non-profit status vs SWR.